CMakeLists.txt

```diff
@@ -158,6 +161,15 @@ if (PREFER_EXTERNAL_DEPS)
       message(WARNING "No cxxopts found in system libraries, using vendored cxxopts...")
     endif()
   endif()
+  # libosmium
+  if (ENABLE_DATA_TOOLS)
+    find_package(Osmium QUIET REQUIRED)
+    if (OSMIUM_FOUND)
+      set(libosmium_include_dir ${OSMIUM_INCLUDE_DIRS})
+    else()
+      message(WARNING "No libosmium found in system libraries, using vendored libosmium...")
+    endif()
+  endif()
```

Comment:
First, the following places should be adjusted as well to make it work

https://github.com/valhalla/valhalla/blob/f87cbbb04cb3bd4e5d5456f3878c8dbff604e190/src/mjolnir/CMakeLists.txt#L93-L94

https://github.com/valhalla/valhalla/blob/18e8959b03153b5512a627ffea6294e15c368c60/test/CMakeLists.txt#L34-L35

Second, it actually looks quite strange to resolve system library only if `ENABLE_DATA_TOOLS` is set... WDYT about going a bit further and define and use `libosmium` cmake target everywhere (to handle dependency on `protozero`) and moreover, build `mjolnir` only if `ENABLE_DATA_TOOLS` is set?